controllers/storagecluster/placement.go

```diff
@@ -17,8 +17,11 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 	if ok {
 		(&in).DeepCopyInto(&placement)
 	} else {
-		in := defaults.DaemonPlacements[component]
-		(&in).DeepCopyInto(&placement)
+		// ignore default PodAntiAffinity mon placements when arbiter is enabled
+		if !(component == "mon" && arbiterEnabled(sc)) {
```

Comment:
This particular change should work. It is not the best way to capture the the different placement requirement for arbiter but the current code does not allow us to make it more readable.

Reply:
I see a panic at line 62 though in my tests:


```
goroutine 601 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x17a88c0, 0x265fd00)
	/usr/lib/golang/src/runtime/panic.go:969 +0x1b9
github.com/openshift/ocs-operator/controllers/storagecluster.getPlacement(0xc000cab200, 0x199ad72, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/controllers/storagecluster/placement.go:62 +0x27f
github.com/openshift/ocs-operator/controllers/storagecluster.newCephCluster(0xc000cab200, 0xc00004203b, 0x1b, 0x4, 0xc000649e60, 0x0, 0x1bc5a00, 0xc0004bf300, 0xc0002ee840)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/controllers/storagecluster/cephcluster.go:284 +0x333
github.com/openshift/ocs-operator/controllers/storagecluster.(*ocsCephCluster).ensureCreated(0x26b2550, 0xc00062e000, 0xc000cab200, 0x0, 0x0)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/controllers/storagecluster/cephcluster.go:106 +0x1065
github.com/openshift/ocs-operator/controllers/storagecluster.(*StorageClusterReconciler).reconcilePhases(0xc00062e000, 0xc000cab200, 0xc0002ee7c0, 0x11, 0xc0002ee760, 0x13, 0x0, 0x0, 0xc000cab200, 0x0)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/controllers/storagecluster/reconcile.go:352 +0x97a
github.com/openshift/ocs-operator/controllers/storagecluster.(*StorageClusterReconciler).Reconcile(0xc00062e000, 0xc0002ee7c0, 0x11, 0xc0002ee760, 0x13, 0xc0004bf200, 0x0, 0x0, 0x0)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/controllers/storagecluster/reconcile.go:167 +0x47b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000d22b40, 0x180c7c0, 0xc0004e6280, 0x0)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:244 +0x2a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000d22b40, 0x203000)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:218 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000d22b40)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:197 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0004bf1f0)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0004bf1f0, 0x1b7edc0, 0xc0008c63f0, 0x1, 0xc000114120)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004bf1f0, 0x3b9aca00, 0x0, 0x203001, 0xc000114120)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc0004bf1f0, 0x3b9aca00, 0xc000114120)
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/home/rtalur/Code/go/src/github.com/openshift/ocs-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:179 +0x416
```

Reply:
yes, I saw that too yesterday while testing in arbiter setup. I'll push the fix today.

Reply:
fixed.